src/components/Table/GridTable.tsx

```diff
@@ -405,25 +410,29 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     // Split out the header rows from the data rows so that we can put an `infoMessage` in between them (if needed).
     const headerRows: RowTuple<R>[] = [];
     const totalsRows: RowTuple<R>[] = [];
-    const filteredRows: RowTuple<R>[] = [];
+    const visibleDataRows: RowTuple<R>[] = [];
+    const matchedRowIds: string[] = [];
 
     // Misc state to track our nested card-ification, i.e. interleaved actual rows + chrome rows
-    const nestedCards = !!style.nestedCards && new NestedCards(columns, filteredRows, style.nestedCards);
+    const nestedCards = !!style.nestedCards && new NestedCards(columns, visibleDataRows, style.nestedCards);
 
-    function visit([row, children]: ParentChildrenTuple<R>, level: number): void {
-      let isCard = nestedCards && nestedCards.maybeOpenCard(row);
-      filteredRows.push([row, makeRowComponent(row, level)]);
+    function visit([row, children]: ParentChildrenTuple<R>, level: number, visible: boolean): void {
+      let isCard = visible && nestedCards && nestedCards.maybeOpenCard(row);
 
-      const isCollapsed = collapsedIds.includes(row.id);
-      if (!isCollapsed && children.length) {
-        nestedCards && nestedCards.addSpacer();
-        visitRows(children, isCard, level + 1);
+      visible && visibleDataRows.push([row, makeRowComponent(row, level)]);
+      matchedRowIds.push(row.id);
```

Comment:
Suggestion: I'm thinking maybe an in-source comment here, like:

```
// This row may be invisible (because it's parent is collapsed), but we still want
// to consider it matched if it or it's parent matched a filter.
```

?